centreon-certified/bsm/bsm-events-apiv2.lua

```diff
@@ -0,0 +1,372 @@
+  local category = self.sc_event.event.category
+  local element = self.sc_event.event.element
+
+  self.sc_logger:debug("[EventQueue:add]: add event in queue category: " .. tostring(self.sc_params.params.reverse_category_mapping[category])
+    .. " element: " .. tostring(self.sc_params.params.reverse_element_mapping[category][element]))
+
+  self.sc_logger:debug("[EventQueue:add]: queue size before adding event: " .. tostring(#self.sc_flush.queues[category][element].events))
+  self.sc_flush.queues[category][element].events[#self.sc_flush.queues[category][element].events + 1] = self.sc_event.event.formated_event
+
+  self.sc_logger:info("[EventQueue:add]: queue size is now: " .. tostring(#self.sc_flush.queues[category][element].events)
+    .. "max is: " .. tostring(self.sc_params.params.max_buffer_size))
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:build_payload, concatenate data so it is ready to be sent
+-- @param payload {string} json encoded string
+-- @param event {table} the event that is going to be added to the payload
+-- @return payload {string} json encoded string
+--------------------------------------------------------------------------------
+function EventQueue:build_payload(payload, event)
+  if not payload then
+    payload = "<event_data>"
+    for index, xml_str in pairs(event) do
+      payload = payload .. "<" .. tostring(index) .. ">" .. tostring(self.sc_common:xml_escape(xml_str)) .. "</" .. tostring(index) .. ">"
+    end
+    payload = payload .. "</event_data>"
+
+  else
+    payload = payload .. "<event_data>"
+    for index, xml_str in pairs(event) do
+      payload = payload .. "<" .. tostring(index) .. ">" .. tostring(self.sc_common:xml_escape(xml_str)) .. "</" .. tostring(index) .. ">"
+    end
+    payload = payload .. "</event_data>"
+  end
+
+  return payload
+end
+
+function EventQueue:send_data(payload)
+  self.sc_logger:debug("[EventQueue:send_data]: Starting to send data")
+
+  -- write payload in the logfile for test purpose
+  if self.sc_params.params.send_data_test == 1 then
+    self.sc_logger:notice("[send_data]: " .. tostring(payload))
+   return true
+  end
+
+  self.sc_logger:info("[EventQueue:send_data]: Going to send the following xml " .. tostring(payload))
+  self.sc_logger:info("[EventQueue:send_data]: BSM Http Server URL is: \"" .. tostring(self.sc_params.params.http_server_url .. "\""))
+
+  local http_response_body = ""
+  local http_request = curl.easy()
+  :setopt_url(self.sc_params.params.http_server_url)
+  :setopt_writefunction(
+    function (response)
+      http_response_body = http_response_body .. tostring(response)
+    end
+  )
+  :setopt(curl.OPT_TIMEOUT, self.sc_params.params.connection_timeout)
+  :setopt(curl.OPT_SSL_VERIFYPEER, self.sc_params.params.allow_insecure_connection)
+  :setopt(
+    curl.OPT_HTTPHEADER,
+    {
+      "Content-Type: text/xml",
+      "content-length: " .. string.len(payload)
+    }
+  )
```

Comment:
```suggestion
  :setopt(curl.OPT_HTTPHEADER, queue_metadata.headers)
```

with the log_curl_command system, headers are already built before. Therefore, we can simplify this line